simulariumio/nerdss/nerdss_converter.py

```diff
@@ -0,0 +1,168 @@
+                    atom_index = atom.index
+                    full_name = resname + "#" + name
+                    position = atom.position
+                    agent_data.types[time_index].append(
+                        TrajectoryConverter._get_display_type_name_from_raw(
+                            full_name, input_data.display_data
+                        )
+                    )
+                    agent_data.unique_ids[time_index][atom_index] = atom.id
+
+                    # Get the user provided display data for this raw_type_name
+                    input_display_data = (
+                        TrajectoryConverter._get_display_data_for_agent(
+                            full_name, input_data.display_data
+                        )
+                    )
+
+                    agent_data.radii[time_index][atom_index] = (
+                        input_display_data.radius
+                        if input_display_data and input_display_data.radius is not None
+                        else 1.0
+                    )
+
+                    # Draw intra-molecular bonds as a fiber between COM (center of mass) and
+                    # binding sites each residue
+                    if name != "ref":
+                        if name == "COM":
+                            # Found the center of mass!
+                            com_pos = list(position)
+                            for bond_site in bond_site_pos:
+                                # Make a fiber to connect COM with bond sites found already
+                                bond_coords = com_pos + bond_site
+                                fiber_positions[time_index].append(bond_coords)
+                        elif com_pos:
+                            # Already found COM, so make a fiber to connect it to this bond site
+                            bond_coords = com_pos + list(position)
+                            fiber_positions[time_index].append(bond_coords)
+                        else:
+                            # Haven't found COM, so keep track of this bond site to connect to COM
+                            # when we find it
+                            bond_site_pos.append(list(position))
+
+        # Add bond data into agent_data
+        bonds_display_data = TrajectoryConverter._get_display_data_for_agent(
+            "bonds", input_data.display_data
+        )
+        next_uid = agent_data.unique_ids.max() + 1
+        for timestep in range(n_timesteps):
+            n_fibers = len(fiber_positions[timestep])
+            n_atoms = int(agent_data.n_agents[timestep])
+            agent_data.n_agents[timestep] = n_fibers + n_atoms
+            for agent_index in range(n_fibers):
+                agent_data.subpoints[timestep][agent_index + n_atoms] = fiber_positions[
+                    timestep
+                ][agent_index]
+                agent_data.n_subpoints[timestep][agent_index + n_atoms] = (
+                    VALUES_PER_3D_POINT * 2
```

Comment:
knowing that all the fibers have only 2 points, it would be interesting if we could try to instance them even more in the viewer.  I'm curious to know if these nerdss trajectories really stress the viewer's ability to render them.  Also I would consider using a cylinder mesh here to get better instancing.  (basically all fibers use the same cylinder as mesh instead of fiber, no subpoints, and you just have to give it the right position and rotation angle...)

Reply:
we could even add cylinder as a primitive in the viewer